### PR TITLE
Trev8939/gen offline map

### PR DESCRIPTION
**Pull request by TADraeseke**
Adds generate offline map sample.

@doneill, one final PR review?

https://devtopia.esri.com/runtime/java-android-sdk/issues/1414

### CODE REVIEW

File: generate-offline-map/README.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+- GenerateOfflineMapResult
+- PortalItem
+- Portal
+- OfflineMapTask
+
+## How it works
+This sample uses the `OfflineMapTask` to take a web map offline.
+
+The sample starts by requesting authentication from the user with the `AuthenticationManager`. On successful authentication, a `PortalItem` is created using a web map ID. The `PortalItem` is then used to initialize an `OfflineMapTask` object.
+
```
Comment (doneill) [inline generate-offline-map/README.md]: Can you explain why authentication is required in the app?

File: generate-offline-map/build.gradle [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        release {
+            minifyEnabled false
+            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
+        }
+    }
+}
+
+dependencies {
+    // arcgis-android & appcompat lib dependencies from rootProject build.gradle
+    compile 'com.android.support:design:25.3.1'
```
Comment (doneill) [inline generate-offline-map/build.gradle]: You can remove this dependency as it is set at the project root level

File: generate-offline-map/src/main/java/com/esri/arcgisruntime/sample/generateofflinemap/MainActivity.java [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+import com.esri.arcgisruntime.portal.Portal;
+import com.esri.arcgisruntime.portal.PortalItem;
+import com.esri.arcgisruntime.security.AuthenticationManager;
+import com.esri.arcgisruntime.security.DefaultAuthenticationChallengeHandler;
+import com.esri.arcgisruntime.symbology.SimpleLineSymbol;
+import com.esri.arcgisruntime.tasks.offlinemap.GenerateOfflineMapJob;
+import com.esri.arcgisruntime.tasks.offlinemap.GenerateOfflineMapParameters;
+import com.esri.arcgisruntime.tasks.offlinemap.OfflineMapTask;
+
+@SuppressWarnings("deprecation") public class MainActivity extends AppCompatActivity {
```
Comment (doneill) [inline generate-offline-map/src/main/java/com/esri/arcgisruntime/sample/generateofflinemap/MainActivity.java]: Why are we suppressing deprecation warnings?

Comment (TADraeseke) [reply generate-offline-map/src/main/java/com/esri/arcgisruntime/sample/generateofflinemap/MainActivity.java]: Progress dialog is now deprecated. I must have hit the button when using the lint inspector.

**Comment by TADraeseke** [comment]
@doneill PR fixes done. Thanks!

**Comment by doneill** [comment]
@TADraeseke Updates look good, merging now.

**Pull request merged**